### ISSUE TITLE
[deploy] Setting Nexus credentials via environment variables does not work

### ISSUE THREAD

**Issue by maciejwalkowiak** [jreleaser/jreleaser#1014]
### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [ ] Example that reproduces the problem (link to git repository is ideal)
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

Releasing Maven project via JReleaser Maven plugin (no other JReleaser config file is created)

1. Add plugin configuration to `pom.xml`:

```xml
<plugin>
  <groupId>org.jreleaser</groupId>
  <artifactId>jreleaser-maven-plugin</artifactId>
  <version>1.3.0</version>
  <configuration>
    <jreleaser>
      <signing>
        <active>ALWAYS</active>
        <armored>true</armored>
      </signing>
      <deploy>
        <maven>
          <nexus2>
            <maven-central>
              <active>ALWAYS</active>
              <url>https://s01.oss.sonatype.org/service/local</url>;
              <closeRepository>true</closeRepository>
              <releaseRepository>true</releaseRepository>
              <stagingRepositories>target/staging-deploy</stagingRepositories>
            </maven-central>
          </nexus2>
        </maven>
      </deploy>
    </jreleaser>
  </configuration>
</plugin>
```

3. Export environment variables:

`export JRELEASER_NEXUS2_USERNAME=...`
`export JRELEASER_NEXUS2_PASSWORD=...`

6. Run `./mvnw install jreleaser:deploy`

### Expected Behaviour

Deploy should happen.

### Actual Behaviour

Maven command fails with:

```
[INFO] --- jreleaser-maven-plugin:1.3.0:deploy (default-cli) @ paseq-maven-plugin ---
[INFO] JReleaser 1.3.0
[INFO]   - basedir set to /Users/maciej/Development/workspaces/playground/maven-jetspeed-plugin/jetspeed-maven-plugin
[INFO] Loading variables from /Users/maciej/.jreleaser/config.properties
[INFO] Validating configuration
[ERROR] == JReleaser ==
[ERROR] nexus2.username must not be blank. Configure a value using the Maven DSL, or define [JRELEASER_NEXUS2_MAVEN_CENTRAL_USERNAME, JRELEASER_NEXUS2_USERNAME] environment variables, or define a key/value pair in /Users/maciej/.jreleaser/config.properties with keys = [JRELEASER_NEXUS2_MAVEN_CENTRAL_USERNAME, JRELEASER_NEXUS2_USERNAME]
[ERROR] nexus2.password must not be blank. Configure a value using the Maven DSL, or define [JRELEASER_NEXUS2_MAVEN_CENTRAL_PASSWORD, JRELEASER_NEXUS2_MAVEN_CENTRAL_TOKEN, JRELEASER_NEXUS2_PASSWORD, JRELEASER_NEXUS2_TOKEN] environment variables, or define a key/value pair in /Users/maciej/.jreleaser/config.properties with keys = [JRELEASER_NEXUS2_MAVEN_CENTRAL_PASSWORD, JRELEASER_NEXUS2_MAVEN_CENTRAL_TOKEN, JRELEASER_NEXUS2_PASSWORD, JRELEASER_NEXUS2_TOKEN]
```

If I define these properties in `~/.jreleaser/config.properties` all works well.

### Environment Information

- **Operating System**: OSX 12.5
- **JReleaser Version:** 1.3.0
- **JDK Version:** OpenJDK Runtime Environment Temurin-17.0.4.1+1 (build 17.0.4.1+1)

Please paste stacktraces from `[out|build|target]/jreleaser/trace.log`.

```
org.jreleaser.model.JReleaserException: JReleaser has not been properly configured.
nexus2.username must not be blank. Configure a value using the Maven DSL, or define [JRELEASER_NEXUS2_MAVEN_CENTRAL_USERNAME, JRELEASER_NEXUS2_USERNAME] environment variables, or define a key/value pair in /Users/maciej/.jreleaser/config.properties with keys = [JRELEASER_NEXUS2_MAVEN_CENTRAL_USERNAME, JRELEASER_NEXUS2_USERNAME]
nexus2.password must not be blank. Configure a value using the Maven DSL, or define [JRELEASER_NEXUS2_MAVEN_CENTRAL_PASSWORD, JRELEASER_NEXUS2_MAVEN_CENTRAL_TOKEN, JRELEASER_NEXUS2_PASSWORD, JRELEASER_NEXUS2_TOKEN] environment variables, or define a key/value pair in /Users/maciej/.jreleaser/config.properties with keys = [JRELEASER_NEXUS2_MAVEN_CENTRAL_PASSWORD, JRELEASER_NEXUS2_MAVEN_CENTRAL_TOKEN, JRELEASER_NEXUS2_PASSWORD, JRELEASER_NEXUS2_TOKEN]

	at org.jreleaser.engine.context.ModelValidator.validate(ModelValidator.java:62)
	at org.jreleaser.workflow.WorkflowImpl.<init>(WorkflowImpl.java:44)
	at org.jreleaser.workflow.Workflows.deploy(Workflows.java:69)
	at org.jreleaser.maven.plugin.JReleaserDeployMojo.execute(JReleaserDeployMojo.java:78)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:370)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:351)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:215)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:171)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:163)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:294)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:960)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:293)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.apache.maven.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:53)
	at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:152)
	at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:76)
```

**Comment by aalmiray** [comment]
The env bar name should include the name of the deployer not just the type, in this the prefix would be `JRELEASER_NEXUS2_MAVEN_CENTRAL_`

**Comment by aalmiray** [comment]
Did you try using the named variants? I'm using `JRELEASER_NEXUS2_MAVEN_CENTRAL_USSERNAME' for example.

**Comment by aalmiray** [comment]
Released in v1.3.1 -> https://github.com/jreleaser/jreleaser/releases/tag/v1.3.1

**Comment by maciejwalkowiak** [comment]
Thanks @aalmiray!
